### DISCUSSION TITLE
Custom recipes with custom ingredients

### DISCUSSION THREAD

**Discussion by 0ffca** [SkriptLang/Skript#5080] [General]
Hi!
Is there an option to create custom recipe with custom ingredients? 
I created skript for it but casual items are also working and it make no sense to use special crafting recipe.
Or maybe is there an option to cancel event while players are trying to craft with vanilla items?

I want to force name tag crafting with specially called ingredients, but skript allows to use vanilla ingredients also.

**Comment by sovdeeth** [comment]
Look at SkBee for custom crafting syntax. Vanilla Skript doesn't include custom recipes.
You can use the `on craft` and `on prepare craft` events to mess with recipes in some capacity, though.

**Reply by 0ffca** [reply]
i used something like this:

on craft:
    if name of event-item is "&f&lZnacznik":
        loop items in player's current inventory:
            if loop-item is feather:
                if name of loop-item is "&f&lPióro pisarskie":
                    stop
    cancel event
    
    
But now I can't craft anything I mean I can't take any item from crafting table. Do you have any idea why?
